test/e2e/specs/editor/various/block-switcher.spec.js

```diff
@@ -8,6 +8,179 @@ test.describe( 'Block Switcher', () => {
+			.click();
+		await page.keyboard.type( '- List content' );
+
+		// Remove blocks.
+		await page.waitForFunction( () => {
+			try {
+				window.wp.data
+					.dispatch( 'core/blocks' )
+					.removeBlockTypes( [
+						'core/quote',
+						'core/pullquote',
+						'core/paragraph',
+						'core/group',
+						'core/heading',
+						'core/columns',
+					] );
+				return true;
+			} catch {
+				return false;
+			}
+		} );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Verify the block switcher isn't enabled.
+		await expect(
+			page
+				.getByRole( 'toolbar', { name: 'Block tools' } )
+				.getByRole( 'button', { name: 'List' } )
+		).toBeDisabled();
+	} );
+
+	test( 'Should show Columns block only if selected blocks are between limits (1-6)', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '- List content' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '## I am a header' );
+		await page.keyboard.down( 'Shift' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.up( 'Shift' );
+
+		await page
+			.getByRole( 'toolbar', { name: 'Block tools' } )
+			.getByRole( 'button', { name: 'Multiple blocks selected' } )
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'menu', { name: 'Multiple blocks selected' } )
+				.getByRole( 'menuitem', { name: 'Columns' } )
+		).toBeVisible();
+	} );
+
+	test( 'Should NOT show Columns transform only if selected blocks are more than max limit(6)', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '- List content' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '## I am a header' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'First paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Third paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Fourth paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Fifth paragraph' );
+		await pageUtils.pressKeys( 'primary+a', { times: 2 } );
+
+		await page
+			.getByRole( 'toolbar', { name: 'Block tools' } )
+			.getByRole( 'button', { name: 'Multiple blocks selected' } )
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'menu', { name: 'Multiple blocks selected' } )
+				.getByRole( 'menuitem', { name: 'Columns' } )
+		).toBeHidden();
+	} );
+
 	test( 'Block variation transforms', async ( { editor, page } ) => {
 		// This is the `stack` Group variation.
```

Comment:
```suggestion
		// This is the `flex` Group variation.
```

This looks like a typo.

Reply:
I don't think so; the variation name is `stack`. The flex is just implementation details.